app/controllers/backend/draft_journals_controller.rb

```diff
@@ -39,7 +39,31 @@ class DraftJournalsController < Backend::BaseController
     end
 
     # this method lists all the entries generated in draft mode
-    def show; end
+    def show
+      @redirection = params[:redirection]
+      @current_page = 1
+      @current_date = Date.today
+      @journal_id = params[:journal_id] ? params[:journal_id].to_i : ''
+      journal_entries = @journal_id == '' ? JournalEntry.all : JournalEntry.where(journal_id: @journal_id)
+      @draft_entries = journal_entries.where(state: :draft).where('printed_on <= ?', @current_date).order(:printed_on)
+      @draft_entries_count = @draft_entries.count
+      @draft_entries = @draft_entries.page(@current_page).per(20)
+      @unbalanced_entries_count = journal_entries.where('printed_on <= ?', @current_date).select { |entry| !entry.balanced? }.count
+      notify_warning_now(:there_are_x_remaining_unbalanced_entries, count: @unbalanced_entries_count) unless @unbalanced_entries_count < 1
+    end
+
+    def list
+      @redirection = params[:redirection]
+      @current_page = params[:page] ? params[:page].to_i : 1
+      @current_date = Date.parse(params[:to]) if params[:to]
+      @journal_id = params[:journal_id] == '' ? params[:journal_id] : params[:journal_id].to_i
```

Comment:
Replace `params[:journal_id] == ''` by `params[:journal_id].blank?` for more readibility ?
Is `params[:journal_id]` always set ?